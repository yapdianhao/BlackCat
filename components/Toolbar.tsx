import React from "react";

import "../styles/Toolbar.scss";
import BlackCatIcon from "../components/BlackCatIcon";
import SideDrawerToggleButton from "./SideDrawerToggleButton";

const profilePic = require("../images/Street-Dance-01.jpg");

interface ToolBarProps {
  drawerClickHandler: React.MouseEventHandler<HTMLButtonElement>;
}

const Toolbar: React.FC<ToolBarProps> = (props) => {
  return (
    <header className="toolbar">
      <nav className="toolbar-navigation">
        <div>
          <SideDrawerToggleButton clickHandler={props.drawerClickHandler} />
        </div>
        <div className="toolbar-logo">
          <BlackCatIcon />
        </div>
        <div>
          <img src={String(profilePic)} className="profile-pic" />
        </div>
      </nav>
    </header>
  );
};

export default Toolbar;
