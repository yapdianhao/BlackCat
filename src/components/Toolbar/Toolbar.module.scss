.toolbar {
  width: 100%;
  position: fixed;
  background: #8560a9;
  top: 0;
  left: 0;
  height: 40px;
}

.toolbarNavigation {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 100%;
  padding: 0 1rem;
}

.toolbarLogo {
  button {
    height: 24px;
    width: 24px;
    fill: #d5ef7f;
  }
}

.profilePic {
  height: 24px;
  width: 24px;
  border-radius: 50%;
}
