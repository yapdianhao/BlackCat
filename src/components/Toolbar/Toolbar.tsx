import React, { useState, useEffect } from "react";
import { useHistory } from "react-router";

import { User } from "../../../server/model/user";
import styles from "./Toolbar.module.scss";
import BlackCatIcon from "../BlackCatIcon/BlackCatIcon";
import SideDrawerToggleButton from "../SideDrawerToggleButton/SideDrawerToggleButton";

interface ToolBarProps {
  drawerClickHandler: React.MouseEventHandler<HTMLButtonElement>;
}

const Toolbar: React.FC<ToolBarProps> = (props) => {
  const history = useHistory();

  const [mainUser, setMainUser] = useState<User>();

  const fetchAuthenticatedUser = async () => {
    await fetch(
      `http://localhost:5000/api/users/${localStorage.getItem(
        "authenticatedUser"
      )}`
    )
      .then((response) => response.json())
      .then((data) => {
        setMainUser(data);
      });
  };

  useEffect(() => {
    fetchAuthenticatedUser();
  }, []);

  return (
    <header className={styles.toolbar}>
      <nav className={styles.toolbarNavigation}>
        <div>
          <SideDrawerToggleButton clickHandler={props.drawerClickHandler} />
        </div>
        <div className={styles.toolbarLogo}>
          <button onClick={() => history.push("/home")}>
            <BlackCatIcon />
          </button>
        </div>
        <div>
          <div onClick={() => history.push("/about")}>
            <img
              src={mainUser && mainUser.userImgUrl}
              className={styles.profilePic}
            />
          </div>
        </div>
      </nav>
    </header>
  );
};

export default Toolbar;
