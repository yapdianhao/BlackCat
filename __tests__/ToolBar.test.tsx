import React from "react";
import renderer from "react-test-renderer";

import Toolbar from "../src/components/Toolbar/Toolbar";

it("renders Toolbar correctly", () => {
  const drawerClickHandler = () => console.log(true);
  const tree = renderer
    .create(<Toolbar drawerClickHandler={drawerClickHandler} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
