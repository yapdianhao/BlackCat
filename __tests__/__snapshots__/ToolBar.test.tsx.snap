// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Toolbar correctly 1`] = `
<header>
  <nav>
    <div>
      <button
        onClick={[Function]}
      >
        <div>
          <svg
            data-name="Layer 1"
            id="Layer_1"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              search
            </title>
            <path
              d="M20.67,18.67H19.61l-0.37-.36a8.68,8.68,0,1,0-.93.93l0.36,0.37v1.05l6.67,6.65,2-2Zm-8,0a6,6,0,1,1,6-6A6,6,0,0,1,12.67,18.67Z"
            />
          </svg>
        </div>
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        <svg
          data-name="Layer 1"
          id="Layer_1"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            logo-cat
          </title>
          <polygon
            points="26.47 14.44 23.07 19.93 23.07 27.38 25.83 29.84 19.2 29.84 21.89 27.36 21.89 19.72 15.69 10.95 19.62 10.95 21.48 9.19 18.18 4.17 14.73 3.14 15.15 -0.03 9.92 4.17 2.83 17.38 7.78 28.12 5.51 30.53 5.51 31.97 9.26 31.97 10.18 31.48 10.93 31.97 29.94 31.97 29.94 30.25 25.68 25.99 25.68 20.55 27.96 16.84 28.78 16.84 29.2 20.08 30.4 20.08 30.71 14.44 26.47 14.44"
          />
        </svg>
      </button>
    </div>
    <div>
      <div
        onClick={[Function]}
      >
        <img />
      </div>
    </div>
  </nav>
</header>
`;
